dkron/job.go

```diff
@@ -303,10 +305,44 @@ func (j *Job) GetTimeLocation() *time.Location {
 	return loc
 }
 
+// scheduleHash replaces H in the cron spec by a value derived from job Name
+// such as "0 0 H * * *"
+func (j *Job) scheduleHash() string {
+	spec := j.Schedule
+	if strings.Contains(spec, "H") && strings.Count(strings.TrimSpace(spec), " ") == 5 {
+		h := 0
+		for _, c := range j.Name {
+			h += int(c)
+		}
+		parts := strings.Split(spec, " ")
+		for index, part := range parts {
+			if strings.Contains(part, "H") {
+				// mods taken in accordance with https://dkron.io/docs/usage/cron-spec/#cron-expression-format
+				ph := h
+				switch index {
+				case 2:
+					ph %= 24
```

Comment:
Not all days are 24 hours long. Not sure how the Jenkins scheduler handled that, but it is worth thinking about and documenting.

Reply:
Not sure I understood? I'm probably missing when are days not 24h long.
Also, that doesn't seem to be an "hash" issue since most people setting up jobs to run at specific hour every day, will choose an hour out of 24...
The reference for these mods are the official dkron documentation for cron expressions, hour part is `0-23`

Reply:
Actually, the day not being 24h long is a bit besides the point - sorry. 

My point was daylight savings time:
When daylight savings time kicks in, for timezones that have that, 2 AM is skipped over: 1:59AM -> 3:00AM. If something is scheduled to jitter around 3AM, it may decide to pick 2AM on a day that does not have 2AM.
The other way around, when coming back from daylight savings time, you have 2AM twice: 2:59AM -> 2:00AM.

This is why it is generally a good idea to run scheduling servers on UTC time. Running a scheduler with a timezone that can jump back and forth in time is risky. Hence, worth thinking about and documenting.

Reply:
Oh nice point on DST! Good stuff to keep in mind for many things!
Personally I have every server in UTC; not just cronjobs, because I never know which apps write logs with UTC or localtime and then depending on the visualization tool, again the same issue, so I'm totally onboard with the UTC recommendation.

That being said, I don't think I agree that it should be worth thinking and documenting "within this PR" but rather at the scheduler level (maybe not even dkron, but the cron lib).
This PR only adds a special character that will resolve to number. And it will resolve for valid values of the field where it's used.
If there are values that might result in bad behavior, it's still the same as if the user would choose that specific value himself (instead of using `H`/`~`).
I agree that we should avoid cases that make sense (like you mentioned for months, hope you agree with using `28` as limit), but for dealing with DST and the issues you mention, "once a day" schedule will be subject to them, regardless of using `0 0 H * * *` or `0 0 3 * * *`.
So I think it's a valid caution/recommendation to note somewhere, but not in the `scheduleHash` specifically, as the risk is not introduced by using it.
What do you think?